apps/l3dtool/l3dtool.cpp

```diff
@@ -781,6 +814,16 @@ int ExtractFile(const Arguments::Extract& args)
 	memcpy(indexBuffer.data.data(), l3d.GetIndices().data(), sizeOfIndices);
 	gltf.buffers.push_back(indexBuffer);
 
+	// TODO(raffclar): tiny_gltf has not yet added support for serialising binary data. Using strings with base64
+	std::map<std::string, tinygltf::Value> extras;
+	extras["footprint"] =
+	    tinygltf::Value(tinygltf::base64_encode(l3d.GetFootprintData().data(), l3d.GetFootprintData().size()));
+	extras["uv2"] = tinygltf::Value(tinygltf::base64_encode(l3d.GetUv2Data().data(), l3d.GetUv2Data().size()));
+	auto& name = l3d.GetNameData();
+	auto nameData = std::vector<unsigned char>(name.begin(), name.end());
+	extras["name"] = tinygltf::Value(tinygltf::base64_encode(nameData.data(), nameData.size()));
```

Comment:
Mesh has a property name that can be used. Wouldn't that be a better place?

Reply:
Also, is name data à string? Why use binary encoding?

Reply:
I want to preserve the size of the name block. I noticed there'sa lot of unused space.

Reply:
Is the size constant with all files?

Reply:
I can't say for certain unless I check every file but so far they're the same. I've only checked a very small subset however

Reply:
If it's always the same size then that's just an issue with their encoding. I think we can have string instead of base64